examples/prefect/runner.py

```diff
@@ -0,0 +1,74 @@
+import sys
+from io import StringIO
+
+import numpy as np
+import training
+from prefect import flow, get_run_logger, task
+from sklearn import base
+
+import nnbench
+from nnbench import reporter, types
+
+
+class PrefectReporter(reporter.ConsoleReporter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.logger = get_run_logger()
+
+    def write(
+        self,
+        record: types.BenchmarkRecord,
+        benchmark_filter: str | None = None,
+        include_context: tuple[str, ...] = (),
+        exclude_empty: bool = True,
+    ) -> None:
+        old_stdout = sys.stdout
+        sys.stdout = StringIO()
```

Comment:
oje, you should never monkey-patch stdlib functions in code.

The way to go here is to attach a console handler to the obtained logger - see the documentation of the logging module on handlers: https://docs.python.org/3/library/logging.html#handler-objects

Reply:
Mhhh that does not seem to work. Prefect supplies a `LoggerAdapter` not a `Logger`. To the former one cannot attach streams.

Reply:
Decided on leaving out formatted reporting. In a real life scenario reports would go into a database and would then be read from a dedicated UI, not the orchestrator. 

I will implement a barebones reporter that simply logs the `dict` returned by the benchmarks to the console though.